tokio-trace/tokio-trace-core/src/dispatcher.rs

```diff
@@ -484,4 +526,65 @@ mod test {
 
         with_default(&Dispatch::new(TestSubscriber), || mk_span())
     }
+
+    #[test]
+    fn global_dispatch() {
+        struct TestSubscriberA;
+        impl Subscriber for TestSubscriberA {
+            fn enabled(&self, _: &Metadata) -> bool {
+                true
+            }
+            fn new_span(&self, _: &span::Attributes) -> span::Id {
+                span::Id::from_u64(1)
+            }
+            fn record(&self, _: &span::Id, _: &span::Record) {}
+            fn record_follows_from(&self, _: &span::Id, _: &span::Id) {}
+            fn event(&self, _: &Event) {}
+            fn enter(&self, _: &span::Id) {}
+            fn exit(&self, _: &span::Id) {}
+        }
+        struct TestSubscriberB;
+        impl Subscriber for TestSubscriberB {
+            fn enabled(&self, _: &Metadata) -> bool {
+                true
+            }
+            fn new_span(&self, _: &span::Attributes) -> span::Id {
+                span::Id::from_u64(1)
+            }
+            fn record(&self, _: &span::Id, _: &span::Record) {}
+            fn record_follows_from(&self, _: &span::Id, _: &span::Id) {}
+            fn event(&self, _: &Event) {}
+            fn enter(&self, _: &span::Id) {}
+            fn exit(&self, _: &span::Id) {}
+        }
+
+        // NOTE: can't have multiple tests that set the global dispatcher
```

Comment:
I think that if these tests were in the integration tests (`tokio-trace-core/tests`), rather than in `src`, you could have multiple tests, if each test was its own separate file in `tests`. This is because each file in `tests` is compiled into its own binary when run, and will thus have its own globals.

We use a similar approach for testing filter caching, since that involves the global registry. You can see that in use [here](https://github.com/tokio-rs/tokio/tree/master/tokio-trace/tests).